### PR TITLE
Enable Dynamic client id Handling

### PR DESCRIPTION
**Pull request by MuhammadAliGhaffarr**
Add support for okta to handle dynamic client id handling

### CODE REVIEW

File: src/oauth2.js [hunk 1]
```diff
@@ -1,38 +1,38 @@
 'use strict';
-const {urlParams} = require('./utils');
```
Comment (mariasyedd) [inline src/oauth2.js:2]: No need to reformat this file

File: src/oauth2.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 	const token_url = config.token_url;
-	const client_id = config.client_id;
+	var client_id = config.client_id;
 	const client_secret = config.client_secret;
-	const redirect_uri = config.redirect_uri;
+	var redirect_uri = config.redirect_uri;
 	const state = config.state;
 	const code_challenge = config.code_challenge;
 	const code_verifier = config.code_verifier;
-	const auth_query_params = config.auth_query_params;
```
Comment (mariasyedd) [inline src/oauth2.js:15]: @MuhammadAliGhaffarr why is this removed?

File: src/oauth2.js [hunk 3]
```diff
@@ -42,79 +42,60 @@ module.exports = function(config)
...
-	{
+	this.begin = (win, callback) => {
 		// Remove menu from the BrowserWindow.
 		win.setMenu(null);
 
 		let authorize_url = `${auth_url}?response_type=code`;
 		authorize_url += `&client_id=${client_id}`;
 		authorize_url += `&scope=${encodeURIComponent(scope)}`;
 		authorize_url += `&redirect_uri=${encodeURIComponent(redirect_uri)}`;
-		if (auth_query_params)
```
Comment (mariasyedd) [inline src/oauth2.js:74]: @MuhammadAliGhaffarr Again, why is this removed?

File: src/oauth2.js [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
+const { urlParams } = require('./utils');
 const request = require('./request');
 
-module.exports = function(config)
-{
+module.exports = function (config) {
 	const auth_url = config.auth_url;
 	const token_url = config.token_url;
-	const client_id = config.client_id;
+	var client_id = config.client_id;
```
Comment (mariasyedd) [inline src/oauth2.js:8]: If this is not reassigned why is it now a var? @MuhammadAliGhaffarr

File: src/oauth2.js [hunk 5]
```diff
@@ -1,38 +1,38 @@
...
 	const client_secret = config.client_secret;
-	const redirect_uri = config.redirect_uri;
+	var redirect_uri = config.redirect_uri;
```
Comment (mariasyedd) [inline src/oauth2.js:10]: If this is not reassigned why is it now a var? @MuhammadAliGhaffarr

**Review by mariasyedd — CHANGES_REQUESTED** [review]

File: src/oauth2.js (hunk 4 above)
Comment (hassanali-pvpn) [reply src/oauth2.js:8]: @MuhammadAliGhaffarr I can understand that client_id will override. But I don't see the code change where it gets reassigned.
